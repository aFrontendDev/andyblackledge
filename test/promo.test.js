import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import Promo from '../app/components/promo';

const testData = {
  "title": "Test Promo",
  "copy": "some text for this promo",
  "id": "id789",
  "link": {
    "href": "#",
    "text": "Link to something"
  }
};

it('should render an empty promo', () => {
  const tree = renderer.create(
    <Promo data={{}} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('should render a promo', () => {
  const tree = renderer.create(
    <Promo data={testData} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});