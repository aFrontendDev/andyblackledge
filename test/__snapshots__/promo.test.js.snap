// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a promo 1`] = `
<div
  className="section-block"
>
  <div
    className="layout layout--a"
  >
    <div
      className="region region--a"
    >
      <div
        className="region-inner"
      >
        <article
          className="block block--size-a promo-test"
          id="id789"
        >
          <h1>
            Test Promo
          </h1>
          <p>
            some text for this promo
          </p>
          <a
            href="#"
          >
            Link to something
          </a>
        </article>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render an empty promo 1`] = `
<div
  className="section-block"
>
  <div
    className="layout layout--a"
  >
    <div
      className="region region--a"
    >
      <div
        className="region-inner"
      >
        <article
          className="block block--size-a promo-test"
          id={undefined}
        >
          <h1 />
          <p />
        </article>
      </div>
    </div>
  </div>
</div>
`;
